pjsip/src/pjsua-lib/pjsua_acc.c

```diff
@@ -4147,24 +4147,43 @@ pj_status_t pjsua_acc_update_contact_on_ip_change(pjsua_acc *acc)
 	       "by IP change", acc->cfg.id.slen,
 	       acc->cfg.id.ptr, (need_unreg ? "un-" : "")));
 
-    status = pjsua_acc_set_registration(acc->index, !need_unreg);
-
-    if ((status != PJ_SUCCESS) && (pjsua_var.ua_cfg.cb.on_ip_change_progress)
+    status = pjsua_acc_set_registration(acc->index, !need_unreg);    
+    if ((status != PJ_SUCCESS)
 	&& (acc->ip_change_op == PJSUA_IP_CHANGE_OP_ACC_UPDATE_CONTACT))
     {
-	/* If update contact fails, notification might already been triggered
-	 * from registration callback.
-	 */
-	pjsua_ip_change_op_info info;
+	if (status == PJSIP_EBUSY) {
+	    /* Retry registration by cancelling the existing one. */
+	    if (acc->regc) {
+		pjsip_regc_destroy(acc->regc);
+		acc->regc = NULL;
+                acc->contact.slen = 0;
+                acc->reg_mapped_addr.slen = 0;
+                acc->rfc5626_status = OUTBOUND_UNKNOWN;
+                acc->rfc5626_flowtmr = 0;
```

Comment:
not quite related with this PR. but perhaps this is time to put this in a separate helper function such as `destroy_regc()` since it has been used throughout pjsua_acc. And I notice a couple of missing of resetting `acc->rfc5626_flowtmr`, which shouldn't happen if we use the helper function.